fiftyone/server/decorators.py

```
@@ -20,43 +21,58 @@
-                    utils.json.serialize(e.sample),
-                    status_code=412,
-                    headers={"ETag": e.etag},
+def route(func=None, *, parse_body: bool = True):
+    """A decorator for HTTPEndpoint methods that handles exceptions.
+
+    By default, it assumes JSON body parsing behavior (for legacy support) and
+    passes parsed data as the third handler argument. Routes without a request
+    body can opt out with ``@route(parse_body=False)``.
+    """
+
+    def decorator(func):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `decorator`

(ANN202)